### ISSUE TITLE
'ansible-galaxy collection build' fails when 'version: null'

### ISSUE THREAD

**Issue by Nihvel** [ansible-collections/cisco.iosxr#236]
https://github.com/ansible-collections/cisco.iosxr/blob/871f68244dae20d800c96def6eb6e59c0aa91569/galaxy.yml#L16

I am using the [RedHat Ansible Automation Hub collection](https://github.com/redhat-cop/ah_configuration) for automating a task that wants to publish collections to the Ansible Automation Hub.

The playbook fails when publishing the collection:
```
TASK [redhat_cop.ah_configuration.publish : Publish Collections] ***************************************************************
failed: [ah00] (item=/var/tmp/cisco.iosxr/cisco-iosxr-*.tar.gz) => {"__ah_collection_file": "/var/tmp/cisco.iosxr/cisco-iosxr-*.tar.gz", "ansible_loop_var": "__ah_collection_file", "changed": false, "msg": "Unable to create collections from /var/tmp/cisco.iosxr/cisco-iosxr-*.tar.gz: Invalid filename cisco-iosxr-*.tar.gz. Expected format: namespace-name-version.tar.gz"}
```

The reason is that `version: null` in the galaxy.yml file will cause the file to be named `cisco-iosxr-*.tar.gz` - where the `*`, which is the `version`, is the error.

The same issue happens also when the command runs manually:
```
(python-venv) [root@virtual cisco.iosxr]# ansible-galaxy collection build /root/juri/aap-config/test/cisco.iosxr --output-path /root/juri/aap-config/test/cisco.iosxr --force
Created collection for cisco.iosxr at /root/juri/aap-config/test/cisco.iosxr/cisco-iosxr-*.tar.gz
```

But when `version: 1`:
```
[root@ah00 cisco.iosxr]# tail -n 1 galaxy.yml 
version: 1
[root@ah00 cisco.iosxr]# ansible-galaxy collection build /var/tmp/cisco.iosxr/ --output-path /var/tmp/cisco.iosxr/ --force 
Created collection for cisco.iosxr at /var/tmp/cisco.iosxr/cisco-iosxr-1.tar.gz
```

The **suggested solution** would be to apply a dummy version info following the semantic version format:
```
version: 0.0.1
```

**Comment by KB-perByte** [comment]
Hey @Nihvel, would be great if you can share some more details about the issue you are facing.
What I can figure out is that you are trying to publish the collection to a private Automation Hub, and if my understanding is right the collection you are willing to publish is a modified version of the collection.
If that is correct the recommendation is to update the galaxy.yml file with the release version which has the custom modifications, as we have it automated in our CI which makes it not a blocker on the collection level.
Please confirm if my understanding is right as per your use case.
Regards

**Comment by Nihvel** [comment]
Hi @KB-perByte, yes that is the issue I noticed. I was trying to publish this collection, as it is, to the Automation Hub. 

I was following the doc in the first post, in particular the README at [https://github.com/redhat-cop/ah_configuration/tree/devel/roles/publish](url) , when I received the error that I wrote above.

I am not actively using this cisco collection, I was just following the doc so I could get *any* collection uploaded. 
I was not able to find a collection that was "Ansible AH ready" so I created a dummy collection respecting every rule the AH imposes and eventually I managed to upload it using the collection I linked in the first post.

In this repo, about the 'version', I read:
> NOTE(pabelanger): We create an empty version key to keep ansible-galaxy 
happy. We dynamically inject version info based on git information.

And since I was not sure if you were aware about this 'issue', that `version: null` makes ansible-galaxy happy and ansible-ah not, maybe a solution could be to use a dummy version like `0.0.1` (AH complains if it is not 'semantic').

Thanks for getting back to me and sorry if it took me a lil bit longer!

**Comment by KB-perByte** [comment]
Hey @Nihvel thanks for the clarity, but were you trying to publish the dummy collection in a private instance of Automation Hub?
As the situation reported is known to us, and we have the process automated for picking the right version number for AH in our CI system, it is not an issue. But for a case where the dummy collection fails you may specify a version number in galaxy.yml. That should fix the issue you are facing.
Regards.

**Comment by Nihvel** [comment]
Hi!
>were you trying to publish the dummy collection in a private instance of Automation Hub?

Yes. I have 2 test instances on different servers. 

And yes, I eventually created in my own git, a dummy collection having the galaxy.yml and meta/runtime.yml properly defined and that let the upload complete.

I do not know why this cisco collection is referenced into that 'AH publish' README I linked before. 
If this is not affiliated in any case to the above project, which makes me think that, maybe, someone took this cisco collection as an example only, and wrote about it on the README, then this issue can closed.

Cheers!
